scripts/e2e-test.sh

```diff
@@ -0,0 +1,226 @@
+
+# Build a DNS testing image with dig
+echo "Building DNS test image with dig..."
+docker build -t dns-test:v1 -f - . <<EOF
+FROM alpine:${ALPINE_VERSION}
+RUN apk add --no-cache bind-tools curl
+ENTRYPOINT ["sh"]
+EOF
+
+# Load all images into kind cluster
+echo "Loading Docker images into kind cluster..."
+kind load docker-image "$EXTERNAL_DNS_IMAGE"
+kind load docker-image dns-test:v1
+
+# Deploy ExternalDNS to the cluster
+echo "Deploying external-dns with custom arguments..."
+
+# Create temporary directory for kustomization
+TEMP_KUSTOMIZE_DIR=$(mktemp -d)
+cp -r kustomize/* "$TEMP_KUSTOMIZE_DIR/"
+
+# Create patch file on the fly
+cat <<EOF > "$TEMP_KUSTOMIZE_DIR/deployment-args-patch.yaml"
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  template:
+    spec:
+      hostNetwork: true
+      containers:
+        - name: external-dns
+          args:
+            - --source=service
+            - --provider=coredns
+            - --txt-owner-id=external.dns
+            - --policy=sync
+            - --log-level=debug
+          env:
+            - name: ETCD_URLS
+              value: http://etcd-0.etcd:2379
+EOF
+
+# Update kustomization.yaml to include the patch
+cat <<EOF > "$TEMP_KUSTOMIZE_DIR/kustomization.yaml"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+images:
+  - name: registry.k8s.io/external-dns/external-dns
+    newName: ${EXTERNAL_DNS_IMAGE%%:*}
+    newTag: ${EXTERNAL_DNS_IMAGE##*:}
+
+resources:
+  - ./external-dns-deployment.yaml
+  - ./external-dns-serviceaccount.yaml
+  - ./external-dns-clusterrole.yaml
+  - ./external-dns-clusterrolebinding.yaml
+
+patchesStrategicMerge:
+  - ./deployment-args-patch.yaml
+EOF
+
+# Apply the kustomization
+kubectl kustomize "$TEMP_KUSTOMIZE_DIR" | kubectl apply -f -
+
+# add a wait for the deployment to be available
+kubectl wait --for=condition=available --timeout=60s deployment/external-dns || true
+
+kubectl describe pods -l app=external-dns
+kubectl describe deployment external-dns
+kubectl logs -l app=external-dns
+
+# Cleanup temporary directory
+rm -rf "$TEMP_KUSTOMIZE_DIR"
+
+# Apply kubernetes yaml with service
+echo "Applying Kubernetes service..."
+kubectl apply -f e2e
+
+# Wait for convergence
+echo "Waiting for convergence (90 seconds)..."
+sleep 90  # normal loop is 60 seconds, this is enough and should not cause flakes
+
+# Check that the records are present
+echo "Checking services again..."
+kubectl get svc -owide
+kubectl logs -l app=external-dns
+
+# Check that the DNS records are present using our DNS server
+echo "Testing DNS server functionality..."
+
+# Get the node IP where the pod is running (since we're using hostNetwork)
+NODE_IP=$(kubectl get nodes -o jsonpath='{.items[0].status.addresses[?(@.type=="InternalIP")].address}')
+echo "Node IP: $NODE_IP"
+
+# Test our DNS server with dig
+echo "Testing DNS server with dig..."
+
+# Create DNS test job that uses dig to query our DNS server
```

Comment:
@Raffo Wdyt about requesting coredns directly ? 
It can be done either by exposing coredns port like exposed in [kind documentation](https://kind.sigs.k8s.io/docs/user/configuration/#extra-port-mappings), or even via port-forward ?